### PR TITLE
RavenDB-11014 add tests and bug fixes

### PR DESCRIPTION
**Pull request by karmeli87**

### CODE REVIEW

File: src/Raven.Client/Http/RequestExecutor.cs [hunk 1]
```diff
@@ -522,7 +522,17 @@ protected void DisposeAllFailedNodesTimers()
 
             if (topologyUpdate != null && topologyUpdate.Status == TaskStatus.RanToCompletion || _disableTopologyUpdates)
             {
-                var (nodeIndex, chosenNode) = ChooseNodeForRequest(command, sessionInfo);
+                int nodeIndex;
+                ServerNode chosenNode;
+                try
+                {
+                    (nodeIndex, chosenNode) = ChooseNodeForRequest(command, sessionInfo);
```
Comment (ppekrol) [inline src/Raven.Client/Http/RequestExecutor.cs]: Better to update the topology in test before doing that, this is very unlikely.

File: src/Raven.Server/Documents/Handlers/IndexHandler.cs [hunk 2]
```diff
@@ -378,7 +378,9 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context.Documents, Re
 
                                     var state = x.State;
 
-                                    if (e.IsOutOfMemory() == false && e.IsRavenDiskFullException() == false)
+                                    if (e.IsOutOfMemory() == false && 
+                                        e.IsRavenDiskFullException() == false && 
+                                        e is OperationCanceledException == false) // we probably closed the indexing thread
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/IndexHandler.cs]: Probably better to handle that ex explicitly and not raise any alerts, etc

File: src/Raven.Client/Http/RequestExecutor.cs (hunk 1 above)
Comment (karmeli87) [reply src/Raven.Client/Http/RequestExecutor.cs]: done

File: src/Raven.Server/Documents/Handlers/IndexHandler.cs (hunk 2 above)
Comment (karmeli87) [reply src/Raven.Server/Documents/Handlers/IndexHandler.cs]: done

**Pull request merged**